packages/babel-block.macro/macro.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+const { createMacro } = require( 'babel-plugin-macros' );
+const { existsSync, readFileSync } = require( 'fs' );
+const { mapKeys, pick } = require( 'lodash' );
+const { dirname, join, relative } = require( 'path' );
+
+function getFilename( [ filenamePath ], state ) {
+	const filename = filenamePath.evaluate().value;
+
+	return join(
+		relative( process.cwd(), dirname( state.file.opts.filename ) ),
+		filename
+	);
+}
+
+function readMetadata( filename ) {
+	if ( ! existsSync( filename ) ) {
+		throw new Error( `Invalid file name provided: ${ filename }.` );
+	}
+
+	const metadataRaw = readFileSync( filename, 'utf8' );
+
+	return JSON.parse( metadataRaw );
+}
+
+function formatMetadata( metadata, types ) {
+	const aliases = {
+		styleVariations: 'styles',
```

Comment:
Is this substitution here for proof of concept, or is this related to some planned change in the block API?

Reply:
It's based on the RFC and we had to do it to remove some confusion with the style files referenced in the same proposal:
[style](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#style)
vs
[styles|styleVariations](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#style-variations)